test/e2e/loadbalancerservice_test.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"sigs.k8s.io/cloud-provider-huaweicloud/pkg/helper"
+	"sigs.k8s.io/cloud-provider-huaweicloud/test/e2e/framework"
+)
+
+var _ = ginkgo.Describe("loadbalancer service testing", func() {
+	var deployment *appsv1.Deployment
+	var service *corev1.Service
+
+	ginkgo.BeforeEach(func() {
+		deploymentName := deploymentNamePrefix + rand.String(RandomStrLength)
+		deployment = helper.NewDeployment(testNamespace, deploymentName)
+		framework.CreateDeployment(kubeClient, deployment)
+	})
+
+	ginkgo.AfterEach(func() {
+		framework.RemoveDeployment(kubeClient, deployment.Namespace, deployment.Name)
+		if service != nil {
+			framework.RemoveService(kubeClient, service.Namespace, service.Name)
+			ginkgo.By(fmt.Sprintf("Wait for the Service(%s/%s) to be deleted", testNamespace, service.Name), func() {
+				gomega.Eventually(func(g gomega.Gomega) (bool, error) {
+					_, err := kubeClient.CoreV1().Services(testNamespace).Get(context.TODO(), service.Name, metav1.GetOptions{})
+					if apierrors.IsNotFound(err) {
+						return true, nil
+					}
+					if err != nil {
+						return false, err
+					}
+					return false, nil
+				}, pollTimeout, pollInterval).Should(gomega.Equal(true))
+			})
+		}
+	})
+
+	ginkgo.It("service enhanced testing", func() {
+		// todo(chengxiangdong): Refactor to use dynamic creation of ELB and VPC later.
+		sessionAffinity := "SOURCE_IP"
+		elbID := "d9b4d06b-8813-46db-bea2-28c5fcbc16f5"
+		subnetID := "82e61d5f-674f-4ef4-a665-d5785a0733e0"
+		lbIP := "192.168.0.55"
```

Comment:
I guess we can start with the service type that auto-create the LB.
`kubernetes.io/elb.autocreate`

Reply:
Great, I have changed to autocreate way.
However, using the existing ELB should also be tested in E2E, which will be added later.